src/sets/signerSet.ts

```diff
@@ -99,381 +96,224 @@ class SignerSet {
   /**
    * addCustody adds a custody address to custodyAdds set.
    *
-   * @param custodyAddress - custody address to add to custodyAdds set
-   */
-  addCustody(custodyAddress: string): Result<void, string> {
-    const sanitizedAddress = this.sanitizeKey(custodyAddress);
-
-    // No-op if address already exists in custodyAdds
-    if (this._custodyAdds.has(sanitizedAddress)) return ok(undefined);
-
-    // Fail if address already exists as a delegate
-    if (this._vertices.has(sanitizedAddress))
-      return err('SignerSet.addCustody: custodyAddress already exists as a delegate');
-
-    // Fail if address has already been removed
-    if (this._custodyRemoves.has(sanitizedAddress)) return err('SignerSet.addCustody: custodyAddress has been removed');
-
-    // Add address to custodyAdds set
-    this._custodyAdds.add(sanitizedAddress);
-    return ok(undefined);
-  }
-
-  /**
-   * removeCustody moves a custody address from the custodyAdds set to the custodyRemoves set.
-   *
-   * @param custodyAddress - custody address to move to custodyRemoves set
+   * @param event - event from Farcaster ID Registry
    */
-  removeCustody(custodyAddress: string): Result<void, string> {
-    const sanitizedAddress = this.sanitizeKey(custodyAddress);
+  addCustody(event: CustodyAddEvent): Result<void, string> {
```

Comment:
writing out the logic as documentation will make this easier to review and update in the future. perhaps something like: 

1. if this address has never been seen before, add it 
2. if this address has been seen and exists in the add set, only add it if it is more recent than the last known one
3. if this address has been seen and exists in the rem set, then find the signer that authorized the removal, and if it is older than the block this address was added in, move it from the rem set to the add set

Reply:
[The notion doc](https://www.notion.so/farcasterxyz/Signer-set-cut-back-ideas-afb8ca2b1d7543a6ad1164425b264c14) has pseudocode, but I can make it simpler and add in the code as a comment as well.